lib/msal-browser/src/interaction_client/BaseInteractionClient.ts

```diff
@@ -178,8 +178,12 @@ export abstract class BaseInteractionClient {
             );
         }
 
-        // Set requested claims hash if claims were requested
-        if (request.claims && !StringUtils.isEmpty(request.claims)) {
+        // Set requested claims hash if claims-based caching is enabled and claims were requested
+        if (
+            this.config.cache.claimsBasedCachingEnabled &&
+            request.claims &&
+            !StringUtils.isEmpty(request.claims)
```

Comment:
```suggestion
            request.claims
```

Reply:
Fixed by switching to isEmptyObj and adding an explanatory comment.